File: komock-core/src/main/kotlin/ua/com/lavi/komock/engine/model/ByteResource.java
package ua.com.lavi.komock.engine.model;

import java.io.ByteArrayInputStream;
import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.net.MalformedURLException;
import java.net.URL;
import java.nio.channels.ReadableByteChannel;

import org.eclipse.jetty.util.resource.Resource;

/**
 * Created by oleksandrloushkin on 25.02.17.
 */
public class ByteResource extends Resource {

    private final byte[] content;

    public ByteResource(byte[] content) {
        this.content = content;
    }

    @Override
    public boolean isContainedIn(Resource resource) throws MalformedURLException {
        return false;
    }

    @Override
    public void close() {
        // nothing to close
    }

    @Override
    public boolean exists() {
        return true;
    }

    @Override
    public boolean isDirectory() {
        return false;
    }

    @Override
    public long lastModified() {
        return 0;
    }

    @Override
    public long length() {
        return 0;
    }

    @Override
    public URL getURL() {
        return null;
    }

    @Override
    public File getFile() throws IOException {
        return null;
    }

    @Override
    public String getName() {
        return null;
    }

    @Override
    public InputStream getInputStream() throws IOException {
        return new ByteArrayInputStream(content);
    }

    @Override
    public ReadableByteChannel getReadableByteChannel() throws IOException {
        return null;
    }

    @Override
    public boolean delete() throws SecurityException {
        return false;
    }

    @Override
    public boolean renameTo(Resource dest) throws SecurityException {
        return false;
    }

    @Override
    public String[] list() {
        return new String[0];
    }

    @Override
    public Resource addPath(String path) throws IOException {
        return null;
    }
}

File: komock-core/src/main/kotlin/ua/com/lavi/komock/engine/model/ByteResource.kt
package ua.com.lavi.komock.engine.model

import org.eclipse.jetty.util.resource.Resource
import java.io.ByteArrayInputStream
import java.io.File
import java.io.InputStream
import java.net.URL
import java.nio.channels.ReadableByteChannel

/**
 * Created by Oleksandr Loushkin
 */

class ByteResource(private val content: ByteArray) : Resource() {

    override fun isContainedIn(resource: Resource): Boolean {
        return false
    }

    override fun close() {
        // nothing to close
    }

    override fun exists(): Boolean {
        return true
    }

    override fun isDirectory(): Boolean {
        return false
    }

    override fun lastModified(): Long {
        return 0
    }

    override fun length(): Long {
        return 0
    }

    override fun getURL(): URL? {
        return null
    }

    override fun getFile(): File? {
        return null
    }

    override fun getName(): String? {
        return null
    }

    override fun getInputStream(): InputStream {
        return ByteArrayInputStream(content)
    }

    override fun getReadableByteChannel(): ReadableByteChannel? {
        return null
    }

    override fun delete(): Boolean {
        return false
    }

    override fun renameTo(dest: Resource): Boolean {
        return false
    }

    override fun list(): Array<String> {
        return arrayOf("")
    }

    override fun addPath(path: String): Resource? {
        return null
    }
}
